packages/gatsby/src/utils/feedback.ts

```diff
@@ -0,0 +1,74 @@
+import report from "gatsby-cli/lib/reporter"
+import { getConfigStore, getGatsbyVersion } from "gatsby-core-utils"
+import latestVersion from "latest-version"
+
+const feedbackKey = `feedback.disabled`
+const lastDateKey = `feedback.lastRequestDate`
+
+// This function is designed to be used by `gatsby feedback --disable`
+// and `gatsby feedback --enable`. This key is used to determine
+// if a user is allowed to be solicited for feedback
+export function setFeedbackDisabledValue(enabled: boolean): void {
+  getConfigStore().set(feedbackKey, enabled)
+}
+
+// Print the feedback request to the user
+export function showFeedbackRequest(): void {
+  getConfigStore().set(lastDateKey, Date.now())
+  report.log(
+    `\n\nHello! Will you help Gatsby improve by taking a four question survey?`
+  )
+  report.log(
+    `Give us your feedback here: https://www.typeform.com/to/A9VWwT\n\n`
+  )
+}
+
+// We are only showing feedback requests to users in if they pass a few checks:
+// 1. They haven't disabled the feedback mechanism
+// 2. They don't have the environment variable to disable feedback present
+// 3. It's been at least 3 months since the last feedback request
+// 4. They are on the most recent version of Gatsby
+export async function userPassesFeedbackRequestHeuristic(): Promise<boolean> {
```

Comment:
Something else that was in my original issue (https://github.com/gatsbyjs/gatsby/issues/14403) is that we want to only show this to a small subset of people each day. We only need ~5 responses a day or so but the heuristics as they stand will show it to a ton of people. This will also have the effect of making survey responses lumpy where we'd rather it be a steady trickle of responses as a lot of people will get the survey and then 3 months later there'll be another lumpy group of responders.

My thought is that we could randomly choose a pretty low % of people e.g. 1 / 250 — and see how many responses we're getting and either lower or raise the % accordingly over time. That would better smooth out how many people are seeing the prompt each day & ensure we're regularly getting feedback.

Reply:
This sounds good to me; @blainekasten I have no objections to this idea!

Reply:
I implemented a fun solution here. Since we want to prompt users every 3 months, i basically built a random number selector for every quarter. Should get us some spread out feedback